### PR TITLE
Product listing

### PR DESCRIPTION
**Pull request by teja538**
added product-listing page
functionality added:
sort by price - high to low and low to high
filter by rating
 
=> looking forward to the suggestions to improve
live demo : https://rad-longma-bee0dc.netlify.app

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,10 +1,27 @@
-import logo from './logo.svg';
+import { Routes, Route } from "react-router-dom";
 import './App.css';
 
+import {Home} from "./pages/home/home.jsx";
+import {Cart} from "./pages/cart/cart.jsx";
+import {WishList} from "./pages/wish-list/wish-list.jsx";
+import {ProductListing} from "./pages/product-listing/product-listing.jsx";
+import {Nav} from "./components/nav.jsx";
```
Comment (Koushith) [inline src/App.js:8]: you can create an index file and export it there. here you can import in a single line.

File: src/back-end/products-db.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+[
+    {
+        "_id":uuid(),
+        "image":ringCat,
+        "name":"Embrace Ring",
+        "subText":"by GIVA jewellery",
+        "rating":4.6,
+        "price":10000,
+        "category":"rings",  
+    },
```
Comment (Koushith) [inline src/back-end/products-db.js:46]: something doesn't look right, key shouldn't be a string😅.

**Review by Koushith — APPROVED** [review]
The Preview link is not accessible. just scanned through the files. overall the file structure looks good.

File: src/App.js [hunk 3]
```diff
@@ -1,10 +1,27 @@
...
+import {Login} from "./pages/auth/login.jsx";
+import {SignUp} from "./pages/auth/sign-up.jsx"
+
```
Comment (Koushith) [inline src/App.js]: you can create an index file and export all components there. and while importing, can use single line import.

File: src/back-end/products-db.js [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+    "image":banglesCat ,
+    },
+    {"_id":uuid(),
+    "categoryName":"Pendants",
+    "image":pendantCat ,
+    },
+    {"_id":uuid(),
+    "categoryName":"Earrings",
+    "image":earringsCat ,
+    },
```
Comment (Koushith) [inline src/back-end/products-db.js:33]: using ``` " " ``` for key's - keys can only be of string type so you can omit ``` " " ```

**Review by Koushith — APPROVED** [review]

File: src/back-end/products-db.js (hunk 4 above)
Comment (Koushith) [reply src/back-end/products-db.js:33]: even though you can use other data types to set an object key, JavaScript automatically converts keys to a string.

File: src/App.js (hunk 3 above)
Comment (teja538) [reply src/App.js]: sorry koushith, I gave direct product_list netlify route so, it's broken I think so. check out this: https://rad-longma-bee0dc.netlify.app/ it's working

File: src/back-end/products-db.js (hunk 4 above)
Comment (teja538) [reply src/back-end/products-db.js:33]: > even though you can use other data types to set an object key, JavaScript automatically converts keys to a string.

got it Koushith! thank you for explaining

File: src/App.js (hunk 3 above)
Comment (teja538) [reply src/App.js]: > you can create an index file and export all components there. and while importing, can use single line import.

yeah will refactor code

File: src/functions/product-list-reducer.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+{
+    if(sortBy === "lowToHigh")
+    {
+        return allProducts.sort((item1,item2) => item1.price-item2.price)
+    }
+    if(sortBy === "highToLow")
+    {
+        return allProducts.sort((item1,item2) => item2.price-item1.price)
+    }
+    console.log("something wrong")
```
Comment (dusariudaykumar) [inline src/functions/product-list-reducer.js:32]: remove ```console.log```

**Review by dusariudaykumar — APPROVED** [review]
The code is clean and your file structure is also good, everything is well structured. Good Work!

File: src/functions/product-list-reducer.js (hunk 5 above)
Comment (teja538) [reply src/functions/product-list-reducer.js:32]: yeah will remove it

**Pull request merged**